### PR TITLE
[213] sendHabitNotification() 404 response when user doesn't exist

### PR DESCRIPTION
**Pull request by ushh789**
For invalid data:
```
{
"email": "1111@gmail.com",
"name": "1111"
}
```
image

### CODE REVIEW

**Comment by ushh789** [comment]
Also, message won't be null, because in other pull requests CustomExceptionHandler is already fixed to provide message of the error

File: service/src/main/java/greencity/service/EmailServiceImpl.java [hunk 1]
```diff
@@ -241,7 +241,11 @@ private void sendEmail(String receiverEmail, String subject, String content) {
     public void sendHabitNotification(String name, String email) {
         String subject = "Notification about not marked habits";
         String content = "Dear " + name + ", you haven't marked any habit during last 3 days";
-        sendEmail(email, subject, content);
+        if (!userRepo.existsUserByEmail(email)) {
+            throw new NotFoundException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL + email);
+        }else {
+            sendEmail(email, subject, content);
+        }
```
Comment (BerserkrMM) [inline service/src/main/java/greencity/service/EmailServiceImpl.java]: Else is not necessary:
`if (!userRepo.existsUserByEmail(email)) {
            throw new NotFoundException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL + email);
        }
        sendEmail(email, subject, content);`

File: service/src/test/java/greencity/service/EmailServiceImplTest.java [hunk 2]
```diff
@@ -140,9 +142,29 @@ void sendRestoreEmailIllegalStateException() {
...
+        service.sendHabitNotification(TestConst.NAME, TestConst.EMAIL);
+
         verify(javaMailSender).createMimeMessage();
+        verify(javaMailSender).send(mimeMessage);
+    }
+
+    @Test
+    @DisplayName("Test sendHabitNotification method when user not found")
+    void sendHabitNotification_userNotFound_throwsNotFoundException() {
+        when(userRepo.findByEmail(TestConst.EMAIL)).thenReturn(Optional.empty());
```
Comment (BerserkrMM) [inline service/src/test/java/greencity/service/EmailServiceImplTest.java]: You should mock existsUserByEmail() instead findByEmail().
Try to verify it:
verify(userRepo).existsUserByEmail(any());
verify(userRepo,never()).findByEmail(any());
You'll see that test passes

File: service/src/main/java/greencity/service/EmailServiceImpl.java (hunk 1 above)
Comment (ushh789) [reply service/src/main/java/greencity/service/EmailServiceImpl.java]: Thanks for the advice, I tried to follow the code that was written before me...
Anyways, I'll clean that "else" in my next commit, thank you for review!

File: service/src/test/java/greencity/service/EmailServiceImplTest.java (hunk 2 above)
Comment (ushh789) [reply service/src/test/java/greencity/service/EmailServiceImplTest.java]: Thanks, I'll change the test in my next commit!

**Review by RockTF — APPROVED** [review]

**Review by RockTF — APPROVED** [review]

**Review by IliaWithHat — APPROVED** [review]

**Review by nataliiapovzhyk — APPROVED** [review]

**Pull request merged**